File: ddd-basic/shop/src/main/java/basic/shop/order/command/domain/value/Address.java
package basic.shop.order.command.domain.value;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.Objects;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@ToString(of = {"zipCode", "address1", "address2"})
public class Address {
    private String zipCode;
    private String address1;
    private String address2;

    @Override
    public boolean equals(Object o) {
        if(o == this)
            return true;
        if(!(o instanceof Address))
            return false;
        Address address = (Address) o;
        return Objects.equals(zipCode, address.zipCode) && Objects.equals(address1, address.address1)
                && Objects.equals(address2, address.address2);
    }

    @Override
    public int hashCode() {
        int result = zipCode == null ? 0 : zipCode.hashCode();
        result = result * 31 + (address1 == null ? 0 : address1.hashCode());
        result = result * 31 + (address2 == null ? 0 : address2.hashCode());
        return result;
    }
}

File: spring-mvc-with-kotlin/order-api/src/main/kotlin/basic/orderapi/common/model/Address.kt
package basic.orderapi.common.model

import javax.persistence.Column
import javax.persistence.Embeddable

@Embeddable
class Address(
    @Column(name = "zip_code")
    var zipCode: String,
    @Column(name = "address1")
    var address1: String,
    @Column(name = "address2")
    var address2: String
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Address

        if (zipCode != other.zipCode) return false
        if (address1 != other.address1) return false
        if (address2 != other.address2) return false

        return true
    }

    override fun hashCode(): Int {
        var result = zipCode.hashCode()
        result = 31 * result + address1.hashCode()
        result = 31 * result + address2.hashCode()
        return result
    }


}